src/codeu/chat/util/CommandTokenizer.java

```diff
@@ -0,0 +1,68 @@
+package codeu.chat.util;
+
+import java.util.ArrayList;
+import java.util.List;
+
+final public class CommandTokenizer {
+
+	private String source;
+	private int currentIndex = 0;
+
+	public boolean hasNextArg() {
+		int remainingIndices = source.length() - currentIndex;
+		return remainingIndices > 0;
+	}
+
+	// returns a String starting at currentIndex until the next occurrence of
+	// endingChar.
+	// Does not include endingChar in the String
+	private String nextStringUntil(char endingChar) {
+		String toReturn = "";
+		while (hasNextArg() && source.charAt(currentIndex) != endingChar) {
+			toReturn += source.charAt(currentIndex);
+			currentIndex++;
+		}
+		currentIndex++;
+		return toReturn;
+	}
+
+	public String getNextArg() {
```

Comment:
I'd also add comments on these functions that explain what they do and what state they leave the class in after they're called. As a general rule of thumb, I'd add JavaDoc to every public function and constructor.